### PR TITLE
fix regression in show_excluded_items

### LINKED ISSUE #190: Wrong use of "show excluded items" in navigation control panel
The new global navigation checks the "show_excluded_items" settings of the Navigation Control Panel, and if the option is unchecked adds "exclude_from_nav=False" to the catalog query to render the main navigation.

This is wrong because the meaning of the "show_excluded_items" option is to *Show items normally excluded from navigation if viewing their children* according to the message that is shown in the control panel.

The exclude_from_nav option should be enforced **always** or change the meaning of the "show_excluded_items" option (and also the default value, because the default is nowadays True)

Here is the control panel definition:

https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/interfaces/controlpanel.py#L830

Here is the check in the global menu viewlet:

https://github.com/plone/plone.app.layout/blob/master/plone/app/layout/viewlets/common.py#L316

Comments @ale-rt ?

### PR DESCRIPTION
**Pull request by petschki**
fixes #190

### CODE REVIEW

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

File: plone/app/layout/viewlets/common.py [hunk 1]
```diff
@@ -324,11 +326,14 @@ def navtree(self):
             'plone.types_use_view_action_in_listings', [])
 
         for brain in brains:
-            brain_path = '/'.join(brain.getPath().split('/'))
+            brain_path = brain.getPath()
```
Comment (petschki) [inline plone/app/layout/viewlets/common.py]: anybody has an idea, why that `split` and `join` combination was there? I just removed it...

Comment (erral) [reply plone/app/layout/viewlets/common.py]: :dancer:

Comment (ale-rt) [reply plone/app/layout/viewlets/common.py]: LOL!

**Comment by ale-rt** [comment]
I agree with @erral in its review and I also thank you for looking at it. Unluckily lately I have been really busy!

File: plone/app/layout/viewlets/common.py [hunk 2]
```diff
@@ -324,11 +326,14 @@ def navtree(self):
...
             brain_parent_path = brain_path.rpartition('/')[0]
             if brain_parent_path == navtree_path:
                 # This should be already provided by the portal_tabs_view
                 continue
+            if not context_path.startswith(brain_path) and brain.exclude_from_nav:  # noqa: E501
```
Comment (mauritsvanrees) [inline plone/app/layout/viewlets/common.py]: Can you switch this condition to first check `brain.exclude_from_nav`? That is a boolean, so the check will be slightly faster than the string comparison.

Comment (petschki) [reply plone/app/layout/viewlets/common.py]: done

**Comment by petschki** [comment]
@mauritsvanrees what do you think about that: https://pyfiddle.io/fiddle/b3173d75-055e-4106-9e85-5509e2be5699/ ... I just could not resist to test this 😄 ... there is a *sliiight* difference, but my example shows, that stringcomparison before boolean is faster. how that?

**Comment by petschki** [comment]
@mauritsvanrees please forgive my incredulty ... my example makes no sense because boolean check is True ... if boolean is False, then the difference is significant when checking boolean first! chapeau!

**Pull request merged**